pkg/pillar/devicenetwork/dns.go

```diff
@@ -149,22 +156,25 @@ func ResolveWithPortsLambda(domain string,
 						defer errsMutex.Unlock()
 						errs = append(errs, err)
 					}
-					if response != nil {
-						resolvedIPsChan <- response
+					if response != nil && len(response) > 0 {
+						select {
+						case resolvedIPsChan <- response:
```

Comment:
Sorry I didn't check your latest changes before they were merged, but this will panic if `resolvedIPsChan` is already closed.
https://go.dev/play/p/I96AZ_2-kcf
It was better when select was used between this channel and reading `quit`, while `resolvedIPsChan` was left for GC.

Reply:
Thx, I will change it.

Reply:
😨😨😨

Reply:
@milan-zededa go back to use `quit` or go back and let the GC close `resolvedIPsChan`?

Reply:
I think we need both but maybe you have simpler solution in mind.

Reply:
Ah, true, I missed this.
This function is not for the faint-hearthed :)

Reply:
No risk, no fun :-)